utils/flink-components-testkit/src/main/scala/pl/touk/nussknacker/engine/flink/util/test/FlinkTestScenarioRunner.scala

```diff
@@ -30,18 +30,43 @@ object testComponents {
   ).map(cd => ComponentDefinition(cd._1, cd._2))
 }
 
-case class SinkForList[T]() extends SinkForType[List[T]]
-
 class FlinkTestScenarioRunner(val components: List[ComponentDefinition], val config: Config, flinkMiniCluster: FlinkMiniClusterHolder) extends ClassBasedTestScenarioRunner {
 
   var testComponentHolder: TestComponentHolder = _
 
-  override def runWithData[I:ClassTag, R](scenario: CanonicalProcess, data: List[I]): RunnerResult[R] = {
+  override def runWithData[I: ClassTag, R](scenario: CanonicalProcess, data: List[I]): RunnerListResult[R] = {
+    implicit val typeInf: TypeInformation[I] = TypeInformation.of(implicitly[ClassTag[I]].runtimeClass.asInstanceOf[Class[I]])
+    val componentsWithData = testComponents.withDataList(data)
+    val componentsWithTestComponents = this.components ++ componentsWithData
+    run(scenario, new TestServiceInvocationCollector(testComponentHolder.runId), Some(testComponentHolder.runId), componentsWithTestComponents).map { _ =>
+      collectResults()
+    }
+  }
+
+  /**
+   * Can be used to test Flink bounded sources - we wait for the scenario to finish.
+   */
+  def runAndCollectResults[R](scenario: CanonicalProcess): RunnerListResult[R] = {
```

Comment:
1. I'm thinking a while if this name is consistent with others. We have runWithData and no CollectingResults suffix. To be consistent it probable should be named just `run` or other methods should be renamed to `runWithDataCollectingResults`. On the other hand just `run` looks a little bit weird and `runWithDataCollectingResults` looks too much verbose. WDYT?
2. `testComponents.withDataList` registers `noopSource` as well. Can you add this component to be consistent with other run methods? It is useful in some cases